homeassistant/components/calendar/__init__.py

```diff
@@ -176,10 +184,36 @@ def _get_date(date):
 
         # cleanup the string so we don't have a bunch of double+ spaces
         self._cal_data['message'] = re.sub('  +', '', summary).strip()
-
         self._cal_data['offset_time'] = offset_time
         self._cal_data['location'] = self.data.event.get('location', '')
         self._cal_data['description'] = self.data.event.get('description', '')
         self._cal_data['start'] = start
         self._cal_data['end'] = end
         self._cal_data['all_day'] = 'date' in self.data.event['start']
+
+
+class CalendarEventView(http.HomeAssistantView):
+    """View to retrieve calendar content."""
+
+    def __init__(self, device):
+        """Initialize calendar view."""
+        self.device = device
+        self.name = "api:calendar" + device.entity_id.split(".")[-1]
```

Comment:
This is wrong, we should not register a view per device. Instead, we need to have 1 view that we pass in the entity component and then can look up using `component.get_entity(entity_id)`.

The url needs to be `/api/calendar/{entity_id}`. [Example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/api.py#L193-L197)

Reply:
Where I should register this view ? in calendar/__init__.py ?

Reply:
In the setup. Camera component does a very similar thing:

 - setup: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/camera/__init__.py#L138
 - view: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/camera/__init__.py#L401-L404

(they do some inheritance things, you don't have to do that)

Reply:
Pushed !
@balloob wow ! This is really clean now !
Thanks !